### ISSUE TITLE
orderlessp/ordergreatp confusing treatment of upper/lower case in symbol names

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#3780]
Imported from SourceForge on 2024-07-08 15:06:09
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2015-09-02 20:10:50
Original: https://sourceforge.net/p/maxima/bugs/3012

---

`orderlessp` and `ordergreatp` order symbol names containing mixed case in an unexpected (non-dictionary) way. This is due to ALPHALESSP (src/clmacs.lisp) which compares symbol names without taking the crazy case inversion rules into account.

I claim that symbols ought to order the same as strings which appear the same as the symbols. But what we get at present:

~~~~
(%i1) orderlessp ("A1", "B1");
(%o1)                                true
(%i2) orderlessp ("A1", "Ba");
(%o2)                                true
(%i3) orderlessp ("Aa", "B1");
(%o3)                                true
(%i4) orderlessp ("Aa", "Ba");
(%o4)                                true
(%i5) orderlessp (A1, B1);
(%o5)                                true
(%i6) orderlessp (A1, Ba);
(%o6)                                false
(%i7) orderlessp (Aa, B1);
(%o7)                                true
(%i8) orderlessp (Aa, Ba);
(%o8)                                true
~~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 15:06:11
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2015-09-02 20:29:01
Original: https://sourceforge.net/p/maxima/bugs/3012/#541e

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 15:06:14
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2015-09-02 20:29:01
Original: https://sourceforge.net/p/maxima/bugs/3012/#ce7e

---

Fixed by commit bd27223. Closing this report.
